src/api/local-storage.js

```diff
@@ -0,0 +1,71 @@
+class localStorageApi {
+
+  static getUsers() {
+    const existingUsers = JSON.parse(localStorage.getItem("storedUser")) || [];
+    return existingUsers;
+  }
+
+  static getUser(userId) {
+    const existingUsers = localStorageApi.getUsers();
+    return existingUsers.find(storedUser =>  storedUser.id === userId );
+  }
+
+  static createUser(newUserProfile) {
+    const existingUsers = localStorageApi.getUsers();
+    existingUsers.push(newUserProfile);
+    localStorage.setItem("storedUser", JSON.stringify(existingUsers));
+    return existingUsers;
+  }
+
+  static updateUser(userId, newUserData) {
+    const existingUsers = localStorageApi.getUsers();
+    const newUserProfile = Object.assign(localStorageApi.getUser(userId), newUserData);
+    const userKey = existingUsers.findIndex(user => user.id === userId);
+    existingUsers.splice(userKey, 1, newUserProfile);
+    localStorage.setItem("storedUser", JSON.stringify(existingUsers));
+  }
+
+  static replaceUser(userId, newUser) {
```

Comment:
what's the difference between `updateUser` and `replaceUser`?

Reply:
`updateUser` is getting the data from the original object and and only changing the specified properties while keeping the original data.

replaceUser` is removing all the properties except id and adding the new specified properties in.

Reply:
so is `replaceUser` changing the id? if so I'm not understanding why you'd want that

Reply:
no,  it's just making sure the id is still attached to the object